cocos/renderer/pipeline/RenderAdditiveLightQueue.cpp

```diff
@@ -47,7 +47,7 @@
 namespace cc {
 namespace pipeline {
 
-RenderAdditiveLightQueue::RenderAdditiveLightQueue(RenderPipeline *pipeline) : _pipeline(static_cast<ForwardPipeline *>(pipeline)),
+RenderAdditiveLightQueue::RenderAdditiveLightQueue(RenderPipeline *pipeline) : _pipeline(dynamic_cast<ForwardPipeline *>(pipeline)),
```

Comment:
why use dynamic_cast?

Reply:
When performing downstream conversions (base class rotor subClass), dynamic_cast has type checking and is safer than static_cast.

Reply:
This is what VisualStudio2019 recommends